File: Sovenire_Collenction_Backend/Models/Artisan.cs
namespace Souvenir_Collection_Backend.Models;

public class Artisan
{
    [Key]
    public Guid Id { get; set; } = Guid.NewGuid();

    public Guid UserId { get; set; }
    public User User { get; set; }

    public string DisplayName { get; set; } = string.Empty;
    public string Region { get; set; } = string.Empty;
    public string CraftType { get; set; } = string.Empty;
    public string Bio { get; set; } = string.Empty;
    public string ProfilePhotoUrl { get; set; } = string.Empty;
    public string ShopAddress { get; set; } = string.Empty;

    [Column(TypeName = "decimal(9,6)")]
    public decimal Lat { get; set; }

    [Column(TypeName = "decimal(9,6)")]
    public decimal Lng { get; set; }

    public bool IsVerified { get; set; } = false;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public ICollection<Product> Products { get; set; }
}

File: Souvenir_Collection_Frontend/lib/models/artisan.dart
// artisan.dart
class Artisan {
  final String id;
  final String name; // maps to DisplayName
  final String region;
  final String craft; // maps to CraftType
  final String bio;
  final String imageUrl; // maps to ProfilePhotoUrl
  final String shopAddress;
  final double lat;
  final double lng;
  final bool isVerified;

  const Artisan({
    this.id = '',
    required this.name,
    this.region = '',
    required this.craft,
    this.bio = '',
    required this.imageUrl,
    this.shopAddress = '',
    this.lat = 0.0,
    this.lng = 0.0,
    this.isVerified = false,
  });

  factory Artisan.fromJson(Map<String, dynamic> json) {
    return Artisan(
      id: json['id']?.toString() ?? '',
      name: json['displayName']?.toString() ?? 'Unknown Artisan',
      region: json['region']?.toString() ?? '',
      craft: json['craftType']?.toString() ?? '',
      bio: json['bio']?.toString() ?? '',
      imageUrl: json['profilePhotoUrl']?.toString() ?? '',
      shopAddress: json['shopAddress']?.toString() ?? '',
      lat: (json['lat'] as num?)?.toDouble() ?? 0.0,
      lng: (json['lng'] as num?)?.toDouble() ?? 0.0,
      isVerified: json['isVerified'] as bool? ?? false,
    );
  }
}